README.md

```diff
@@ -497,83 +496,71 @@ client.stats(function(err, statsPage) {        // statsPage as PaginatedResult
 client.time(function(err, time) { ... }); // time is in ms since epoch
 ```
 
-## Test suite
-
-To run both the NodeUnit & Karma Browser tests, simply run the following command:
-
-    npm test
-
-## NodeUnit Tests
-
-Run the NodeUnit test suite
-
-    npm run test:nodeunit
-
-Or run just one or more test files
-
-    npm run test:nodeunit -- --test spec/realtime/auth.test.js
+## Using the async API style
 
-## Browser Tests
+### Realtime Example
 
-Browser tests are run using [Karma test runner](http://karma-runner.github.io/0.12/index.html).
+```ts
+import * as Ably from 'ably/promises';
 
-### To build & run the tests in a single step
+const client = new Ably.Realtime.Promise(options);
 
-    npm run test:karma
+const ablyRealtimePromiseExample = async () => {
+  const channel = client.channels.get('myChannel');
 
-### Debugging the tests in your browser with NodeUnit test runner
+  // Attaching to a channel
+  await channel.attach();
 
-Simply open [spec/nodeunit.html](./spec/nodeunit.html) in your browser to run the test suite with a nice GUI.
+  client.connection.on('connected', async () => {
+    // Getting presence on a channel
```

Comment:
```
  // Attaching to a channel
  await channel.attach();

  client.connection.on('connected', async () => {
    // Getting presence on a channel
```

So, I know your intentions here are just listing a bunch of stuff that people can do, in no particular order. Trouble is, when people see a block of code in a readme, they tend to take it as a starting point, copy and paste the whole thing, and modify it. And if they do that with this, they'll get something broken (because it awaits the attach and then adds a connected listener once the channel is attached, which will then never fire because by the time the channel is attached it's already connected).

So while we can definitely do an example code block that lists a bunch of stuff you can do, I think that stuff has to be in an order that will actually work if run in that order.

(TBH I would get rid of the `on('connected'` listener from the basic example block. Unless there's something someone specifically wants to do every time the library reconnects, which is relatively rare, there's no actual need to do stuff within the connected listener, including it here might give a misleading impression that it's required)

Reply:
Thanks, I can see that the `connection.on` was pointless so I've removed that and made the examples _pretty much_ independently "runnable". The only real caveat is that the realtime example requires that the clientId has been specified (otherwise they throw at `channel.presence.enter()`). I'm not sure if it's worth making that obvious somehow or assuming that developers will generally be able to figure that out for themselves...

Reply:
👍 much better